apps/swirl-docs/src/pages/sitemap.xml.ts

```diff
@@ -23,17 +23,21 @@ function SiteMap() {
   // getServerSideProps will do the heavy lifting
 }
 
-export async function getServerSideProps({ res }: any) {
-  const paths: string[] = [];
-
+function addNavItemsToPaths(paths: string[], navItems: NavItem[]) {
   navItems.forEach((item) => {
```

Comment:
i know forEach modifies the passed in array, but i do not like this implicit behaviour. i think it would be better to use map and use a return value. in that way we can directly call this function in line 38 and just return the array directly instead of creating the const and manipulating it through reference